### PR TITLE
for Bohdan/

### PR DESCRIPTION
**Pull request by vlykhvar**

### CODE REVIEW

File: src/main/java/com/dev/theater/confing/AppConfing.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
+
+@Configuration
+@PropertySource("classpath:db.properties")
+@ComponentScan(basePackages = {
+        "com.dev.theater.service",
+        "com.dev.theater.dao"
```
Comment (Alex-Korzun) [inline src/main/java/com/dev/theater/confing/AppConfing.java]: I've got comment that we should separate Authentication Services in additional package `Security` and also add it here to @ComponentScan

File: src/main/java/com/dev/theater/controller/HelloController.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping
+public class HelloController {
+
+    @ResponseBody
```
Comment (Alex-Korzun) [inline src/main/java/com/dev/theater/controller/HelloController.java]: If we're using @RestController there is no more need in @ResponseBody. It has already this annotation inside.

File: src/main/java/com/dev/theater/dao/impl/TicketDaoImpl.java [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
-import com.dev.theater.library.Dao;
 import com.dev.theater.model.Ticket;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
 
-@Dao
+@Repository
 public class TicketDaoImpl extends DaoImpl<Ticket> implements TicketDao {
+    @Autowired
```
Comment (Alex-Korzun) [inline src/main/java/com/dev/theater/dao/impl/TicketDaoImpl.java:11]: If you use @Autowired annotation, you should use it everywhere or don't use it. It's up to you.
P.S.: you missed it in `CinemaHallDaoImpl`

File: src/main/java/com/dev/theater/service/impl/CinemaHallImpl.java [hunk 4]
```diff
@@ -1,17 +1,22 @@
...
 import com.dev.theater.model.CinemaHall;
 import com.dev.theater.service.CinemaHallService;
 import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
 @Service
 public class CinemaHallImpl implements CinemaHallService {
-    @Inject
+
```
Comment (Alex-Korzun) [inline src/main/java/com/dev/theater/service/impl/CinemaHallImpl.java]: Don't think there should be empty line. And in further impls as well.

File: src/main/java/com/dev/theater/service/impl/CinemaHallImpl.java [hunk 5]
```diff
@@ -1,17 +1,22 @@
...
     private CinemaHallDao cinemaHallDao;
```
Comment (Alex-Korzun) [inline src/main/java/com/dev/theater/service/impl/CinemaHallImpl.java]: It should be final. As in further impls

**Review by Alex-Korzun — CHANGES_REQUESTED** [review]

**Review by Alex-Korzun — APPROVED** [review]

File: pom.xml [hunk 6]
```diff
@@ -7,18 +7,57 @@
...
+        <dependency>
+            <groupId>javax.servlet.jsp.jstl</groupId>
+            <artifactId>jstl-api</artifactId>
+            <version>1.2</version>
+        </dependency>
+        <dependency>
+            <groupId>taglibs</groupId>
+            <artifactId>standard</artifactId>
+            <version>1.1.2</version>
+        </dependency>
```
Comment (KseniiaMakarova) [inline pom.xml]: do we need these dependencies?

File: pom.xml [hunk 7]
```diff
@@ -7,18 +7,57 @@
...
+            <artifactId>servlet-api</artifactId>
+            <version>2.5</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet.jsp</groupId>
+            <artifactId>javax.servlet.jsp-api</artifactId>
+            <version>2.3.3</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (KseniiaMakarova) [inline pom.xml]:
```suggestion
        <dependency>
            <groupId>javax.servlet</groupId>
            <artifactId>javax.servlet-api</artifactId>
            <version>3.0.1</version>
            <scope>provided</scope>
        </dependency>
```

File: src/main/java/com/dev/theater/controller/HelloController.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
+package com.dev.theater.controller;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping
+public class HelloController {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/controller/HelloController.java]: redundant empty line

File: src/main/java/com/dev/theater/controller/HelloController.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package com.dev.theater.controller;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/controller/HelloController.java]: what is the purpose of this annotation here?

File: src/main/java/com/dev/theater/dao/impl/DaoImpl.java [hunk 10]
```diff
@@ -2,17 +2,27 @@
...
 import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
 
+@Repository
 public class DaoImpl<T> implements GenericDao<T> {
+    protected final SessionFactory sessionFactory;
+
+    @Autowired
+    public DaoImpl(SessionFactory sessionFactory) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/dao/impl/DaoImpl.java:14]: let's make this class abstract and see the following article
https://www.baeldung.com/spring-autowired-abstract-class#constructor-injection

File: src/main/java/com/dev/theater/dao/impl/UserDaoImpl.java [hunk 11]
```diff
@@ -2,18 +2,25 @@
...
 import java.util.Optional;
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
 
-@Dao
+@Repository
 public class UserDaoImpl extends DaoImpl<User> implements UserDao {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/dao/impl/UserDaoImpl.java]: redundant empty line

File: src/main/java/com/dev/theater/security/impl/AuthenticationServiceImpl.java [hunk 12]
```diff
@@ -1,22 +1,26 @@
...
 
 @Service
 public class AuthenticationServiceImpl implements AuthenticationService {
-    @Inject
     private UserService userService;
-    @Inject
     private ShoppingCartService shoppingCartService;
 
+    @Autowired
+    AuthenticationServiceImpl(UserService userService, ShoppingCartService shoppingCartService) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/security/impl/AuthenticationServiceImpl.java]: is something missing?

File: src/main/java/com/dev/theater/service/impl/CinemaHallImpl.java [hunk 13]
```diff
@@ -1,17 +1,22 @@
...
 
+    @Autowired
+    CinemaHallImpl(CinemaHallDao cinemaHallDao) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/service/impl/CinemaHallImpl.java]: same for all ServiceImpl classes

File: src/main/java/com/dev/theater/service/impl/CinemaHallImpl.java [hunk 14]
```diff
@@ -1,17 +1,22 @@
...
-import com.dev.theater.library.Inject;
-import com.dev.theater.library.Service;
 import com.dev.theater.model.CinemaHall;
 import com.dev.theater.service.CinemaHallService;
 import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
 @Service
 public class CinemaHallImpl implements CinemaHallService {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/service/impl/CinemaHallImpl.java]:
```suggestion
public class CinemaHallServiceImpl implements CinemaHallService {
```

File: src/main/java/com/dev/theater/service/impl/MovieSessionServiceImpl.java [hunk 15]
```diff
@@ -1,18 +1,23 @@
...
 import java.time.LocalDate;
 import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
 @Service
 public class MovieSessionServiceImpl implements MovieSessionService {
-    @Inject
+
     private MovieSessionDao movieSessionDao;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/service/impl/MovieSessionServiceImpl.java]: it is better to make `final` all the dependencies that we autowire on service and dao layers

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
let's remove `hibernate.cfg.xml` as well

File: src/main/java/com/dev/theater/dao/impl/DaoImpl.java [hunk 16]
```diff
@@ -2,17 +2,25 @@
...
 import com.dev.theater.dao.GenericDao;
 import com.dev.theater.exception.CrudException;
-import com.dev.theater.util.HibernateUtil;
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
 
+@Repository
 public class DaoImpl<T> implements GenericDao<T> {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/dao/impl/DaoImpl.java]: can we make the class abstract?

File: pom.xml [hunk 17]
```diff
@@ -7,18 +7,46 @@
...
         <maven.checkstyle.plugin.configLocation>
             https://raw.githubusercontent.com/mate-academy/style-guides/master/java/checkstyle.xml
         </maven.checkstyle.plugin.configLocation>
     </properties>
     <dependencies>
+        <dependency>
+            <groupId>javax.servlet.jsp.jstl</groupId>
+            <artifactId>jstl-api</artifactId>
+            <version>1.2</version>
+        </dependency>
```
Comment (KseniiaMakarova) [inline pom.xml]: remove this

File: pom.xml [hunk 18]
```diff
@@ -34,6 +62,16 @@
...
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-dbcp2</artifactId>
+            <version>2.8.0</version>
+        </dependency>
+        <dependency>
+            <groupId>taglibs</groupId>
+            <artifactId>standard</artifactId>
+            <version>1.1.2</version>
+        </dependency>
```
Comment (KseniiaMakarova) [inline pom.xml]: same

File: pom.xml [hunk 19]
```diff
@@ -7,18 +7,46 @@
...
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+            <version>${spring.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>servlet-api</artifactId>
+            <version>2.5</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (KseniiaMakarova) [inline pom.xml]:
```suggestion
<dependency>
            <groupId>javax.servlet</groupId>
            <artifactId>javax.servlet-api</artifactId>
            <version>3.0.1</version>
            <scope>provided</scope>
        </dependency>
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/theater/confing/AppConfing.java [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
+
+@Configuration
+@PropertySource("classpath:db.properties")
+@ComponentScan(basePackages = {
+        "com.dev.theater.service",
+        "com.dev.theater.dao",
+        "com.dev.theater.security"
+})
+public class AppConfing {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/confing/AppConfing.java]:
```suggestion
public class AppConfig {
```

File: src/main/java/com/dev/theater/confing/MyWebAppInitializer.java [hunk 21]
```diff
@@ -0,0 +1,20 @@
+package com.dev.theater.confing;
+
+import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;
+
+public class MyWebAppInitializer extends AbstractAnnotationConfigDispatcherServletInitializer {
+    @Override
+    protected Class<?>[] getRootConfigClasses() {
+        return new Class[]{AppConfing.class};
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/confing/MyWebAppInitializer.java]:
```suggestion
        return new Class[]{AppConfig.class};
```

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**